core/sql/cli/Globals.cpp

```diff
@@ -187,18 +187,9 @@ void CliGlobals::init( NABoolean espProcess,
 
   ComRtGetProcessPriority(myPriority_);
   savedPriority_ = (short)myPriority_;
-  myNumSegs_ = 0;
-  myNumCpus_ = 0;
-  SEGMENT_INFO * segs = new(&executorMemory_) SEGMENT_INFO[MAX_NO_OF_SEGMENTS];
-  ComRtGetSegsInfo(segs, MAX_NO_OF_SEGMENTS, myNumSegs_,
-           (NAHeap *)&executorMemory_);
-  for (Lng32 i = 0; i < myNumSegs_; i++)
-    {
-      myNumCpus_ += segs[i].noOfCpus_;
-    }
-  NADELETEARRAY(segs, MAX_NO_OF_SEGMENTS, SEGMENT_INFO, &executorMemory_);
+  myNumCpus_ = ComRtGetCPUArray(cpuArray_, (NAHeap *)&executorMemory_);
```

Comment:
Good question, what do you think? I'm a bit scared to assert, because that could make the whole system inaccessible. Could we log an error event instead? But I'm not sure whether that is possible at the time when we create the CLI globals, maybe the event log code is dependent on something that has not been created yet. I'll check and add either an assert or a log event.